ground/src/main/java/com/google/android/ground/system/LocationManager.kt

```diff
@@ -44,43 +44,70 @@ private val FINE_LOCATION_UPDATES_REQUEST =
 class LocationManager
 @Inject
 constructor(
+  @ApplicationScope private val externalScope: CoroutineScope,
   private val permissionsManager: PermissionsManager,
   private val settingsManager: SettingsManager,
-  private val locationClient: RxFusedLocationProviderClient
+  private val locationClient: RxFusedLocationProviderClient,
+  private val mapStateRepository: MapStateRepository
 ) {
 
-  private val locationUpdates: @Hot(replays = true) Subject<Location> = BehaviorSubject.create()
-  private val locationUpdateCallback: RxLocationCallback = RxLocationCallback(locationUpdates)
+  val locationLockState: MutableStateFlow<Result<Boolean>> =
+    MutableStateFlow(Result.success(mapStateRepository.isLocationLockEnabled))
+  private val locationUpdatesStateFlow: MutableStateFlow<Location?> = MutableStateFlow(null)
```

Comment:
Does this stream contain location updates (i.e., events) or the current location (i.e., states)? If it's the former, this should probably be modeled as a Channel.

If they're states, I'd name them accordingly. In that case we can drop the `StateFlow` suffix for better readability, imo.

Reply:
I'm not sure I fully understand the distinction, between location updates vs current location. I think I'm leaning towards the latter because it receives updates to the location from the StateFlowCallback, but it stores the current location as state, which is then combined with the locationLockState in the locationLockAwareLocationUpdates (poorly named I think) to provide the current location if the location lock is enabled, and null if not.

I'll go ahead and update the names of both the locationUpdatesStateFlow and locationLockAwareLocationUpdates to currentLocationState and locationLockAwareCurrentLocation.

Let me know if that makes sense, or if you still think we should use a channel instead of a StateFlow.

Reply:
SG, removed unnecessary qualifications and we're now exposing the location updates through a Channel.